### PR TITLE
Fix `KeyError` as only worker 0 gets TPU_CONFIG

### PR DESCRIPTION
**Pull request by jysohn23**

### CODE REVIEW

File: torch_xla/distributed/xla_multiprocessing.py [hunk 1]
```diff
@@ -202,7 +202,7 @@ def _prepare_env_for_index(index, pf_cfg):
   os.environ[xenv.LOCAL_ORDINAL] = str(index)
 
   if pf_cfg.dev_kind == 'TPU':
-    _setup_tpu_worker(index, gindex, pf_cfg, os.environ[xenv.TPU_CONFIG])
+    _setup_tpu_worker(index, gindex, pf_cfg, os.environ.get(xenv.TPU_CONFIG, None))
```
Comment (dlibenzi) [inline torch_xla/distributed/xla_multiprocessing.py:205]: I'd add an explicit `assert` at line 188, to check that `tpu_env_config` is not `None`.

Comment (jysohn23) [reply torch_xla/distributed/xla_multiprocessing.py:205]: You mean, in addition to this change right?

Comment (dlibenzi) [reply torch_xla/distributed/xla_multiprocessing.py:205]: > You mean, in addition to this change right?

Yes, of course 😄

Comment (jysohn23) [reply torch_xla/distributed/xla_multiprocessing.py:206]: Done, thanks!

File: torch_xla/distributed/xla_multiprocessing.py [hunk 2]
```diff
@@ -185,6 +185,7 @@ def _setup_tpu_worker(index, gindex, pf_cfg, tpu_env_config):
   if xenv.LOCAL_WORKER not in os.environ:
     # The local worker can be missing for a 1 TPU host setup. Make sure we
     # always have one.
+    assert (tpu_env_config is not None), 'tpu_env_config must not be None'
```
Comment (dlibenzi) [inline torch_xla/distributed/xla_multiprocessing.py:188]: You do not need parenthesis there 😉

Comment (jysohn23) [reply torch_xla/distributed/xla_multiprocessing.py]: Sure, removed.

**Pull request merged**